### PR TITLE
Enhances FeatureManager to throw an exception for missing feature filters.

### LINKED ISSUE #13: An exception should be thrown if a configured feature filter has not been registered.
If a feature is configured to be enabled for a certain feature filter, the feature management pipeline should throw if that feature is being evaluated and the feature filter can not be found. Without this behavior it is very hard to track down the fact that a feature filter has been excluded from being registered. For those who intend for this behavior and depend on an exception not being thrown, this could be configurable via feature management options.

### PR DESCRIPTION
**Pull request by jimmyca15**
The current feature manager implementation only logs a warning when a feature filter is missing for a given feature. This makes missing feature filters tricky to diagnose and can lead to unintended behavior. This enhancement changes the default behavior of the feature manager so that it will throw a `FeatureManagementException` when it encounters a missing feature filter.

This new behavior can be disabled by configuring the feature management options. 

```
services.Configure<FeatureManagementOptions>(options =>
{
    options.IgnoreMissingFeatureFilters = true;
});
```

This resolves #13 

@codapus-co

### CODE REVIEW

File: src/Microsoft.FeatureManagement/FeatureManagementOptions.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+namespace Microsoft.FeatureManagement
+{
+    /// <summary>
+    /// Options that control the behavior of the feature management system.
+    /// </summary>
+    public class FeatureManagementOptions
+    {
+        /// <summary>
+        /// Controls the behavior of feature evaluation when dependent feature filters are missing.
+        /// If missing features filters are not ignored an exception will be thrown when attempting to evaluate a feature that depends on a missing feature filter.
```
Comment (zhenlan) [inline src/Microsoft.FeatureManagement/FeatureManagementOptions.cs]: NIT: ... feature~s~ filters ...

Comment (jimmyca15) [reply src/Microsoft.FeatureManagement/FeatureManagementOptions.cs]: Thanks ! I'll update before merge.

File: src/Microsoft.FeatureManagement/FeatureManager.cs [hunk 2]
```diff
@@ -21,15 +22,22 @@ class FeatureManager : IFeatureManager
...
+            ILoggerFactory loggerFactory,
+            IOptions<FeatureManagementOptions> options)
         {
             _settingsProvider = settingsProvider;
             _featureFilters = featureFilters ?? throw new ArgumentNullException(nameof(featureFilters));
             _sessionManagers = sessionManagers ?? throw new ArgumentNullException(nameof(sessionManagers));
             _logger = loggerFactory.CreateLogger<FeatureManager>();
             _filterMetadataCache = new ConcurrentDictionary<string, IFeatureFilterMetadata>();
             _contextualFeatureFilterCache = new ConcurrentDictionary<string, ContextualFeatureFilterEvaluator>();
+            _options = options?.Value ?? throw new ArgumentNullException(nameof(options));
```
Comment (MSGaryWang) [inline src/Microsoft.FeatureManagement/FeatureManager.cs]: >options?.Value [](start = 23, length = 14)

so we don't expect a default value?

Comment (jimmyca15) [reply src/Microsoft.FeatureManagement/FeatureManager.cs]: The default value should be returned by Value.

**Pull request merged**